homeassistant/components/roomba/binary_sensor.py

```diff
@@ -52,23 +45,8 @@ def icon(self):
     @property
     def state(self):
         """Return the state of the sensor."""
-        return self._bin_status
-
-    @property
-    def device_info(self):
-        """Return the device info of the vacuum cleaner."""
-        return {
-            "identifiers": {(DOMAIN, self._identifier)},
-            "name": str(self._name),
-        }
-
-    async def async_update(self):
-        """Return the update info of the vacuum cleaner."""
-        # No data, no update
-        if not self.vacuum.master_state:
-            _LOGGER.debug("Roomba %s has no data yet. Skip update", self.name)
-            return
-        self._bin_status = (
+        bin_status = (
             roomba_reported_state(self.vacuum).get("bin", {}).get("full", False)
         )
-        _LOGGER.debug("Update Full Bin status from the vacuum: %s", self._bin_status)
+        _LOGGER.debug("Update Full Bin status from the vacuum: %s", bin_status)
```

Comment:
Side effects are not allowed in entity state properties. Please remove the logging.